### DISCUSSION TITLE
Field "image" must not have a selection since type "String" has no subfields.

### DISCUSSION THREAD

**Discussion by jf-home** [gatsbyjs/gatsby#33939] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I'm getting ERROR #85922  GRAPHQL with a `gatsby develop` and `gatsby build`, originally stated under https://github.com/gatsbyjs/gatsby/issues/33878#issue-1046266146 which has since been closed, so resubmitting under a new ticket.  

Minimum reproduction available as indicated below...

### Reproduction Link

https://github.com/jf-home/gatsby4-images

### Steps to Reproduce

1. Clone the repo
2. npm i
3. gatsby develop
...
gatsby clean
gatsby develop

### Expected Result

Develop/build session to complete with no errors.

### Actual Result

 ERROR #85922  GRAPHQL

There was an error in your GraphQL query:

      Field "image" must not have a selection since type "String" has no subfields.

      This can happen if you e.g. accidentally added { } to the field "image". If you didn't expect "image" to be of type "String" make sure that your input source and/or plugin is correct.
      However, if you expect "value" to exist, the field might be accessible in another subfield. Please try your query in GraphiQL and use the GraphiQL explorer to see  
which fields you can query and what shape they have.

      It is recommended to explicitly type your GraphQL schema if you want to use optional fields. This way you don't have to add the mentioned
      "dummy content". Visit our docs to learn how you can define the schema for "undefined":
      https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization#creating-type-definitions

File: src\pages\index.js:30:15

failed extract queries from components - 2.667s

### Environment

```shell
System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
  Binaries:
    Node: 14.15.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.8 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.9.0
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (95.0.1020.44)
  npmPackages:
    gatsby: ^4.1.2 => 4.1.2 
    gatsby-background-image: ^1.5.3 => 1.5.3 
    gatsby-plugin-catch-links: ^4.1.0 => 4.1.0 
    gatsby-plugin-google-fonts: ^1.0.1 => 1.0.1 
    gatsby-plugin-html-attributes: ^1.0.5 => 1.0.5 
    gatsby-plugin-image: ^2.1.0 => 2.1.0 
    gatsby-plugin-manifest: ^4.1.1 => 4.1.1 
    gatsby-plugin-netlify: ^3.14.0 => 3.14.0 
    gatsby-plugin-netlify-cms: ^6.1.0 => 6.1.0 
    gatsby-plugin-offline: ^5.1.1 => 5.1.1 
    gatsby-plugin-react-helmet: ^5.1.0 => 5.1.0 
    gatsby-plugin-sharp: ^4.1.1 => 4.1.1 
    gatsby-remark-images: ^6.1.1 => 6.1.1 
    gatsby-remark-relative-images: ^2.0.2 => 2.0.2 
    gatsby-source-filesystem: ^4.1.0 => 4.1.0 
    gatsby-transformer-remark: ^5.1.1 => 5.1.1 
    gatsby-transformer-remark-frontmatter: ^1.1.0 => 1.1.0 
    gatsby-transformer-sharp: ^4.1.0 => 4.1.0 
  npmGlobalPackages:
    gatsby-cli: 2.12.114
```

**Comment by graysonhicks** [comment]
This looks like an issue with `gatsby-transformer-remark-frontmatter` and `gatsby-remark-relative-images`.  I was able to clone and resolve by:

- removing those plugins from package.json, gatsby-node and gatsby-config
- remove this filesystem setting from gatsby-config:
```
   {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `assets`,
        path: `${__dirname}/static/assets`,
      },
    },
```
- move the 'circuit' image to `src/images`
- change the index.md frontmatter to `../images/circuit.jpg`

Then both `develop` and `build` work and the image is visible in the browser.  I believe those plugins are dated and unnecessary.

**Reply by jf-home** [reply]
Hi @graysonhicks - thanks for the reply - much appreciated!

The proposed solution is not viable as I'm using Netlify CMS for the frontend data entry which sets image references to _/assets/circuit.png_ (located under the static folder) or _/src/images/circuit.png_ for frontmatter image fields.

Netlify CMS won't save an image path like _../images/circuit.png._

So, I need a solution that works with relative images, hence the inclusion of the `gatsby-transformer-remark-frontmatter` and `gatsby-remark-relative-images` plugins.

The reason for holding images under the static folder is to simplify the path construct for content editors when they are including images in the markdown body (html region). This way, images references should be stored as _/assets/circuit.png_ and get resolved to the full path at develop and build time by the plugins mentioned above.

Thanks again for your time with this.  Please let me know if you have any further suggestions.

**Reply by davemullenjnr** [reply]
I'm running into this same issue today after migrating from Gatsby v3 to v4. I also use Netlify CMS. It was working fine before the migration. Wondering if you managed to find a solution to this? 🤞

**Comment by jf-home** [comment]
Hi there!

It was a while ago when I had this issue so I can’t remember specifically
what the issue was.  If I recall correctly, everything was OK with the
plug-ins installed and their order in gatsby-config.js.  I think the issue
was more that I had a null value being returned for an image from one of my
graphql queries.  Once I tracked that down and corrected it, all fell into
place as it should.

I hope that helps.
